### ISSUE TITLE
Express/Typescript template fails to start

### ISSUE THREAD

**Issue by gatesn** [remix-run/remix#6119]
### What version of Remix are you using?

1.15.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

`npx create-remix@latest`, select express adapter, select Typescript, run `npm run dev` 

### Expected Behavior

The dev server to startup

### Actual Behavior

```
[nodemon] starting `cross-env NODE_ENV=production node ./server.js --require dotenv/config ./server.js`
node:internal/modules/cjs/loader:1078
  throw err;
  ^

Error: Cannot find module '/.../my-remix-app/server.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.15.0
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by andrew-dias** [comment]
The `start` command also doesn't work since it references the non-existent `server.js`. It seems like it is missing the steps to build and bundle the server code. Taking inspiration from the Blues stack template, I currently have my `scripts` section as follows and it seems to work:
```
"scripts": {
    "build": "run-s build:*",
    "build:remix": "remix build",
    "build:server": "esbuild --platform=node --format=cjs ./server.ts --outdir=build --bundle",
    "dev": "run-p dev:*",
    "dev:build": "cross-env NODE_ENV=development npm run build:server -- --watch",
    "dev:remix": "cross-env NODE_ENV=development remix watch",
    "dev:server": "cross-env NODE_ENV=development node --inspect --require ./node_modules/dotenv/config ./build/server.js",
    "start": "cross-env NODE_ENV=production node ./build/server.js",
    "typecheck": "tsc"
  }
```

**Comment by BUGO07** [comment]
> The `start` command also doesn't work since it references the non-existent `server.js`. It seems like it is missing the steps to build and bundle the server code. Taking inspiration from the Blues stack template, I currently have my `scripts` section as follows and it seems to work:
> 
> ```
> "scripts": {
>     "build": "run-s build:*",
>     "build:remix": "remix build",
>     "build:server": "esbuild --platform=node --format=cjs ./server.ts --outdir=build --bundle",
>     "dev": "run-p dev:*",
>     "dev:build": "cross-env NODE_ENV=development npm run build:server -- --watch",
>     "dev:remix": "cross-env NODE_ENV=development remix watch",
>     "dev:server": "cross-env NODE_ENV=development node --inspect --require ./node_modules/dotenv/config ./build/server.js",
>     "start": "cross-env NODE_ENV=production node ./build/server.js",
>     "typecheck": "tsc"
>   }
> ```

yeah but this is using javascript for server.js..
would be nice to have express with ts

**Comment by andrew-dias** [comment]
> yeah but this is using javascript for server.js.. would be nice to have express with ts

Not really, the `build:server` command generates the `build/server.js` file from `server.ts`.

**Comment by BUGO07** [comment]
> > yeah but this is using javascript for server.js.. would be nice to have express with ts
> 
> Not really, the `build:server` command generates the `build/server.js` file from `server.ts`.

oh, i tried this now and it works great except for one thing, the dev script doesn't update the server after i change server.ts, it does say it rebuilds it into server.js but changes aren't applie unless i restart

**Comment by BUGO07** [comment]
> > > yeah but this is using javascript for server.js.. would be nice to have express with ts
> > 
> > 
> > Not really, the `build:server` command generates the `build/server.js` file from `server.ts`.
> 
> oh, i tried this now and it works great except for one thing, the dev script doesn't update the server after i change server.ts, it does say it rebuilds it into server.js but changes aren't apply unless i restart

changing the dev:server script to use nodemon instead of node can fix this :]

**Comment by peterbe** [comment]
I solved this locally by adding 
```
npm i @digitak/esrun 
npm i dotenv
```

Then, in `server.ts` I added:
```js
import dotenv from "dotenv";

dotenv.config();
```

and in `package.json` made these changes:

```diff
diff --git a/package.json b/package.json
index dc239b6..af8d710 100644
--- a/package.json
+++ b/package.json
@@ -4,9 +4,9 @@
   "scripts": {
     "build": "remix build",
     "dev": "npm-run-all build --parallel \"dev:*\"",
-    "dev:node": "cross-env NODE_ENV=development nodemon --require dotenv/config ./server.js --watch ./server.js",
+    "dev:node": "cross-env NODE_ENV=development esrun --watch ./server.ts ",
     "dev:remix": "remix watch",
-    "start": "cross-env NODE_ENV=production node ./server.js",
+    "start": "cross-env NODE_ENV=production esrun ./server.ts",
     "typecheck": "tsc",
     "lint": "eslint app",
     "lintfix": "eslint app --fix && prettier --write app",

```

Seems to work. Doesn't generate a `server.js` file.

**Comment by dteknolove** [comment]
Waiting a an update from the official remix team 🙏

**Comment by lili21** [comment]
https://github.com/remix-run/remix/pull/6145

this one should fix the issue

**Comment by dpobel** [comment]
for the record, the typecheck script also fails right the install

```
npm run typecheck

> typecheck
> tsc

node_modules/@remix-run/node/dist/upload/fileUploadHandler.d.ts:47:22 - error TS2420: Class 'NodeOnDiskFile' incorrectly implements interface 'File'.
  Property 'prototype' is missing in type 'NodeOnDiskFile' but required in type 'File'.

47 export declare class NodeOnDiskFile implements File {
                        ~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2572:5
    2572     prototype: Blob;
             ~~~~~~~~~
    'prototype' is declared here.


Found 1 error in node_modules/@remix-run/node/dist/upload/fileUploadHandler.d.ts:47
```
